packages/medusa/src/services/product-variant.js

```diff
@@ -558,14 +558,11 @@ class ProductVariantService extends BaseService {
       }
 
       query.where = qb => {
-        qb.where(where).andWhere(
-          new Brackets(qb => {
-            qb.where([
-              { sku: Raw(a => `${a} ILIKE :q`, { q: `%${q}%` }) },
-              { title: Raw(a => `${a} ILIKE :q`, { q: `%${q}%` }) },
-            ]).orWhere(`product.title ILIKE :q`, { q: `%${q}%` })
-          })
-        )
+        qb.where(where).andWhere([
+          { sku: ILike(`%${q}%`) },
+          { title: ILike(`%${q}%`) },
+          { product: ILike(`%${q}%`) },
```

Comment:
Should this not be `{ product: { title: ILike(`%${q}%`) } }`? Just looking at the other ones above :)

Reply:
And then delete the inner join?

Reply:
Nice catch! Sorry, will fix now!

Reply:
We should consider adding a couple of integration tests to ensure it functions :)